it/v1/endpoints/RetrieveTransactionDetailsControllerISpec.scala

```diff
@@ -0,0 +1,170 @@
+        response.json shouldBe mtdPaymentJson
+      }
+    }
+
+    "return a 404 NO_TRANSACTIONS_FOUND error" when {
+      "a success response with no data is returned" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+          DesStub.onSuccess(DesStub.GET, desUrl, OK, desJsonNoTransactions)
+        }
+
+        val response: WSResponse = await(request.get)
+
+        response.status shouldBe NOT_FOUND
+        response.header("Content-Type") shouldBe Some("application/json")
+        response.json shouldBe Json.toJson(NoTransactionDetailsFoundError)
+      }
+    }
+
+    "return error according to spec" when {
+
+      def validationErrorTest(requestNino: String, fromDate: String,
+                              toDate: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+        s"validation fails with ${expectedBody.code} error" in new Test {
+
+          override val nino: String = requestNino
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request.get)
+          response.status shouldBe expectedStatus
+          response.json shouldBe Json.toJson(expectedBody)
+          response.header("Content-Type") shouldBe Some("application/json")
+        }
+      }
+
+      val input = Seq(
+        ("AA1123A", "2018-10-01", "2019-10-01", BAD_REQUEST, NinoFormatError)
+      )
+
+      input.foreach(args => (validationErrorTest _).tupled(args))
+    }
+
+    "des service error" when {
+
+      def errorBody(code: String): String =
+        s"""{
+           |  "code": "$code",
+           |  "reason": "des message"
+           |}""".stripMargin
+
+      def serviceErrorTest(desStatus: Int, desCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+        s"des returns an $desCode error and status $desStatus" in new Test {
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+            DesStub.onError(DesStub.GET, desUrl, desStatus, errorBody(desCode))
+          }
+
+          val response: WSResponse = await(request.get)
+          response.status shouldBe expectedStatus
+          response.json shouldBe Json.toJson(expectedBody)
+          response.header("Content-Type") shouldBe Some("application/json")
+        }
+      }
+
+      val input = Seq(
+        (BAD_REQUEST, "INVALID_IDTYPE", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+        (BAD_REQUEST, "INVALID_REGIME_TYPE", INTERNAL_SERVER_ERROR, DownstreamError),
+        (BAD_REQUEST, "NOT_FOUND", NOT_FOUND, NotFoundError),
```

Comment:
The **des status code** for **NotFoundError** is **NOT_FOUND** not **BAD_REQUEST**